### PR TITLE
R1 Final Review

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

**Comment by owlot** [comment]
## Documents
- [ ] ICD
  - [ ] Current interfaces (grpc & rest)
  - [ ] Current relevant document links
- [ ] SDD
  - [ ] Description of grpc & rest handler implementations
  - [ ] Current sequence diagrams where applicable
  - [ ] Current relevant document links
- [ ] CONOPS
  - [ ] Current information
  - [ ] Bibliography (if any citations)
- [ ] README (see svc-cargo for example)
  - [ ] Badges
  - [ ] Accurate installation instructions
  - [ ] Current links to other documents
  - [ ] Arrow DAO links

## Code (not applicable to examples/ folder)
- [ ] No unhandled unwrap()s
- [ ] Accurate (!) code comments on all functions
- [ ] No `# Arguments` and `# Returns` in function comments
- [ ] Function arguments checked where possible
- [ ] Sufficient logger (info, warn, and debug) messages
- [ ] Sensible Types (unsigned vs. signed)

## REST API
- [ ] Return codes accurately documented with openapi
- [ ] Return codes make sense
- [ ] Sufficiently descriptive error messages

## Cargo.toml(s)
- [ ] Package versions correct
- [ ] No minor versions specified
- [ ] Links to crates.io package, not git branch

**Comment by owlot** [comment]
From reading the RFC, I think the 409 `CONFLICT` status codes should be replaced with a 500 `internal server error` as it's not a user input error that would result in the grpc client to return a error code.

**Comment by ServiceDog** [comment]
> From reading the RFC, I think the 409 `CONFLICT` status codes should be replaced with a 500 `internal server error` as it's not a user input error that would result in the grpc client to return a error code.

Fixed: https://github.com/Arrow-air/svc-cargo/pull/16/commits/78af236c3900962fe594cab7257bee784fba08a3

**Comment by ServiceDog** [comment]
> There are still some function with # Arguments and # Returns in their comments. If you want to fix this in R2, I'm fine with that as well, but it would be nice to have a ticket for it in that case.

Fixed: https://github.com/Arrow-air/svc-cargo/pull/16/commits/78af236c3900962fe594cab7257bee784fba08a3

**Comment by owlot** [comment]
All looks good!
One last request is to change the version in the Cargo.toml files to v0.1.0 so we'll have a proper changelog generated when merging to `main`.

File: client-grpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[package]
+edition = "2021"
+name    = "svc-cargo-client-grpc"
+version = "0.6.0-develop.1"
```
Comment (owlot) [inline client-grpc/Cargo.toml:4]: Please reset the version to "0.1.0" so the release workflow won't run into issues and we get a proper CHANGELOG generated

File: server/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+[package]
+name    = "svc-cargo"
+version = "0.6.0-develop.1"
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/Cargo.toml]: https://github.com/Arrow-air/svc-cargo/pull/16/commits/bbc5a4b421abcb5923ad3701a58f8af7c5e682b4

File: client-grpc/Cargo.toml (hunk 1 above)
Comment (ServiceDog) [reply client-grpc/Cargo.toml]: https://github.com/Arrow-air/svc-cargo/pull/16/commits/bbc5a4b421abcb5923ad3701a58f8af7c5e682b4

**Review by owlot — APPROVED** [review]

File: server/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+log                       = "0.4"
+openssl                   = "0.10"
+prost                     = "0.11"
+prost-build               = "0.11"
+prost-types               = "0.11"
+serde                     = "1.0"
+serde_json                = "1.0"
+svc-pricing-client-grpc   = { git = "https://github.com/Arrow-air/svc-pricing" }
+svc-scheduler-client-grpc = { git = "https://github.com/Arrow-air/svc-scheduler" }
+svc-storage-client-grpc   = { git = "https://github.com/Arrow-air/svc-storage" }
```
Comment (owlot) [inline server/Cargo.toml:28]: Will all need to specifically point to the main branch once these services are all released.
So we can't merge before that.

**Review by owlot — CHANGES_REQUESTED** [review]

File: server/Cargo.toml (hunk 3 above)
Comment (ServiceDog) [reply server/Cargo.toml]: Updated, will rerun CI when svc-scheduler gets merged in

**Review by owlot — APPROVED** [review]

**Pull request merged**